pil-analyzer/src/untyped_evaluator.rs

```diff
@@ -17,7 +17,9 @@ pub fn evaluate_expression_to_int(
     expr: parsed::Expression,
 ) -> Result<BigInt, EvalError> {
     evaluator::evaluate_expression::<GoldilocksField>(
-        &ExpressionProcessor::new(driver, &Default::default()).process_expression(expr),
+        &ExpressionProcessor::new(driver, &Default::default())
+            .process_expression(expr)
+            .map_err(|e| EvalError::TypeError(e.message().to_string()))?, //TODO: better error type
```

Comment:
It's necessary to add a new variant to EvalError for this case. What name, generic enough to be a “processing error”. should we give it?

Reply:
I mean in the end, this function should return two different errors. I think it would be better to change this function to just take an `analyzed::Expression`, and do the expression processing at the caller side.

Three of the four call sites are inside StatementProcessor which makes regular use of Expression processor anyway.